src/main/java/com/commercetools/project/sync/category/CategorySyncer.java

```diff
@@ -80,22 +78,14 @@ public static CategorySyncer of(
 
     final CustomObjectService customObjectService = new CustomObjectServiceImpl(targetClient);
 
-    final CategoryReferenceTransformService referenceTransformService =
-        new CategoryReferenceTransformTransformServiceImpl(sourceClient);
-
-    return new CategorySyncer(
-        categorySync,
-        sourceClient,
-        targetClient,
-        customObjectService,
-        referenceTransformService,
-        clock);
+    return new CategorySyncer(categorySync, sourceClient, targetClient, customObjectService, clock);
   }
 
   @Override
   @Nonnull
   protected CompletionStage<List<CategoryDraft>> transform(@Nonnull final List<Category> page) {
-    return this.referencesService.transformCategoryReferences(page);
+    final ReferenceIdToKeyCache referenceIdToKeyCache = new CaffeineReferenceIdToKeyCacheImpl();
+    return toCategoryDrafts(getSourceClient(), referenceIdToKeyCache, page);
```

Comment:
I found that `throwable` is handled in `ProductSyncer` after coverting to draft.  Would you align the approach here with `ProductSyncer`?

Reply:
ProductSyncer has attributes reference replacement logic where we are throwing the exception and it is handled. But for other Syncers , We don't have any exception thrown.